curriculum/challenges/_meta/lecture-working-with-data-fetching-and-memoization-in-react/meta.json

```diff
@@ -0,0 +1,23 @@
+{
+  "name": "Working with Data Fetching and Memoization in React",
+  "blockType": "lecture",
+  "blockLayout": "challenge-list",
+  "isUpcomingChange": true,
+  "dashedName": "lecture-working-with-data-fetching-and-memoization-in-react",
+  "superBlock": "full-stack-developer",
+  "challengeOrder": [
```

Comment:
we need to add the useOptimistic hook script 

What is the useOptimistic hook and how does it work?